units/stats_pod.go

```diff
@@ -0,0 +1,104 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/pod"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const podStatsCollectorJobName = "pod-stats-collector"
+
+func init() {
+	registry.AddJobType(podStatsCollectorJobName, func() amboy.Job {
+		return makePodStatsCollector()
+	})
+}
+
+type podStatsCollector struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+}
+
+// NewPodStatsCollector logs statistics about current pod usage.
+func NewPodStatsCollector(id string) amboy.Job {
+	j := makePodStatsCollector()
+	j.SetID(fmt.Sprintf("%s.%s", podStatsCollectorJobName, id))
+	j.SetEnqueueScopes(podStatsCollectorJobName)
+
+	return j
+}
+
+func makePodStatsCollector() *podStatsCollector {
+	j := &podStatsCollector{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    podStatsCollectorJobName,
+				Version: 0,
+			},
+		},
+	}
+	return j
+}
+
+func (j *podStatsCollector) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		j.AddError(err)
+		return
+	}
+
+	if flags.BackgroundStatsDisabled {
+		grip.Debug(message.Fields{
+			"mode":     "degraded",
+			"impact":   "pod stats collection disabled",
+			"job":      j.ID(),
+			"job_type": j.Type().Name,
+		})
+		return
+	}
+
+	j.AddError(j.logStats())
+}
+
+func (j *podStatsCollector) logStats() error {
+	statuses := []pod.Status{
```

Comment:
If we are hardcoding the list of statuses to check should we just pass no parameters to `GetStatsByStatus` and define the list there instead?

Reply:
Or do we plan on potentially using this with different inputs in the future? Not a CR, just a suggestion.

Reply:
I mostly did this so I could zero out tha values below this in the log message and use the same statuses between the find and the log messages. We might change the inputs, but not that frequently.